connection.go

```
@@ -0,0 +1,405 @@
+			},
+		),
+		nats.ReconnectHandler(
+			func(nc *nats.Conn) {
+				natsConnectionID, err := c.generateNatsConnectionID()
+				if err != nil {
+					memphisKafkaErr("error reconnecting to memphis cost")
+				}
+
+				clientReconnectionUpdateReq := ClientReconnectionUpdateReq{
+					NewNatsClientID: natsConnectionID,
+					ClientID:        c.ClientID,
+				}
+
+				clientReconnectionUpdateReqBytes, err := json.Marshal(clientReconnectionUpdateReq)
+				if err != nil {
+					memphisKafkaErr("error reconnecting to memphis cost")
+				}
+
+				_, err = nc.Request(clientReconnectionUpdateSubject, clientReconnectionUpdateReqBytes, 5*time.Second)
+				if err != nil {
+					memphisKafkaErr("error reconnecting to memphis cost")
+				}
+
+				c.NatsClientID = natsConnectionID
+			},
+		),
+	}
+
+	nc, err := nats.Connect(host, opts...)
+	if err != nil {
+		return fmt.Errorf("memphis_kafka: error connecting to memphis cost")
+	}
+
+	js, err := nc.JetStream()
+	if err != nil {
+		return fmt.Errorf("memphis_kafka: error connecting to memphis cost")
+	}
+
+	c.BrokerConnection = nc
+	c.JSContext = js
+
+	natsConnectionID, err := c.generateNatsConnectionID()
+	if err != nil {
+		return fmt.Errorf("memphis_kafka: error connecting to memphis cost")
+	}
+	c.NatsClientID = natsConnectionID
+
+	return nil
+}
+
+func (c *Client) RegisterClient() error {
+	registerReq := RegisterReq{
+		NatsClientID: c.NatsClientID,
+		Language:     "go",
+	}
+
+	registerReqBytes, err := json.Marshal(registerReq)
+	if err != nil {
+		return fmt.Errorf("memphis_kafka: error registering client")
+	}
+
+	resp, err := c.BrokerConnection.Request(clientRegisterSubject, registerReqBytes, 5*time.Second)
+	if err != nil {
+		return fmt.Errorf("memphis_kafka: error registering client")
+	}
+
+	var registerResp RegisterResp
+	err = json.Unmarshal(resp.Data, &registerResp)
+	if err != nil {
+		return fmt.Errorf("memphis_kafka: error registering client")
+	}
+
+	if registerResp.Err != "" {
```

I meant that max connection error should be returned from nats in this case not a custom message from the backend, so once this kind of error occurs you should raise this error